### PR TITLE
Add declarative argocd setup docs

### PR DESCRIPTION
**Pull request by alexmt**

### CODE REVIEW

File: docs/declarative-setup.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+## Manage Argo CD using Argo CD
+
+Argo CD is able to manage itself since all settings are represented by Kubernetes manifests. The suggested way is to create [Kustomize](https://github.com/kubernetes-sigs/kustomize)
+based application which uses base Argo CD manifests from https://github.com/argoproj/argo-cd and apply required changes on top.
+
+Example of `kustomization.yaml`:
+
+```yaml
+bases:
+- github.com/argoproj/argo-cd//manifests/cluster-install?ref=v0.10.0
```
Comment (dthomson25) [inline docs/declarative-setup.md]: You have an extra / in url: `argo-cd//manifests`

File: docs/declarative-setup.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    certData: string
+    # Server should be accessed without verifying the TLS certificate
+    insecure: boolean
+    # PEM-encoded bytes (typically read from a client certificate key file).
+    keyData: string
+    # ServerName is passed to the server for SNI and is used in the client to check server
+    # ceritificates against. If ServerName is empty, the hostname used to contact the
+    # server is used.
+    serverName: string
+
```
Comment (dthomson25) [inline docs/declarative-setup.md]: Nitpick: extra newline here

File: docs/declarative-setup.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+resources:
+- clusters-secrets.yaml
+- repos-secrets.yaml
+
+# changes to config maps
+patchesStrategicMerge:
+- overlays/argo-cd-cm.yaml
+```
+
+The live example of self managed Argo CD config is available at https://cd.apps.argoproj.io and with configuration
```
Comment (dthomson25) [inline docs/declarative-setup.md]: Should we add a note about how you will need to sign in with your github account to access the live Argo CD?

Also, nitpick: Make https://cd.apps.argoproj.io into a hyperlink allowing users to click it.

File: docs/declarative-setup.md (hunk 1 above)
Comment (dthomson25) [reply docs/declarative-setup.md]: Also, should we change the ref to the latest version: v0.10.6?

**Review by dthomson25 — APPROVED** [review]
I have a couple small questions, but otherwise it looks great to me!

File: docs/declarative-setup.md (hunk 2 above)
Comment (alexmt) [reply docs/declarative-setup.md]: fixed

File: docs/declarative-setup.md (hunk 1 above)
Comment (alexmt) [reply docs/declarative-setup.md]: changed to v0.10.6 . After v0.11.0 release lets introduce moving tag `stable` and use it in all docs

Comment (alexmt) [reply docs/declarative-setup.md]: The extra / is intentional. Kustomize uses it to indicate end of repo url and start of relative path inside

File: docs/declarative-setup.md (hunk 3 above)
Comment (alexmt) [reply docs/declarative-setup.md]: Added note. Github renders URLs as hyperlinks automatically.

**Pull request merged**